### DISCUSSION TITLE
How to sample full architectures from defined Search Spaces?

### DISCUSSION THREAD

**Discussion by geantrindade** [microsoft/nni#4212] [General]
I want to sample random architectures from the defined Search Spaces such as DARTS and ENAS. 

Is it possible to sample the entire PyTorch object representing the whole architecture? If not, is it possible to sample all the possible cells from the search spaces? Can you provide some code examples for doing this? 

I found some code examples from ENAS and DARTS but the model returned include all the "layer choices" that I assume the Trainer would select then generate the final architecture, but I am more interested in just sample random architectures from the search spaces, I mean, the full architectures without these "decisions to be made" included. Can you pleaseeee help me with that?

**Comment by ultmaster** [comment]
You can use random strategy in multi-trial NAS, which will sample one architecture in each trial.

**Comment by geantrindade** [comment]
Could you show me some code examples of how to do that? I just want to sample some architectures or cells from DARTS actually...

**Comment by geantrindade** [comment]
I tried to use the Random strategy but the documentation says that "Do not dry run to get the full search space. Used when the search space has variational size or candidates" which is not the case for the defined search spaces such as DARTS. Could you please give me an example of how to simply sample at random the cells or entire architectures from DARTS space? I'm still waiting...

**Comment by ultmaster** [comment]
I don't think DARTS search space has a variational number of candidates.

You can refer to multi-trial/mnist example on how to use Random strategy.

**Comment by scarlett2018** [comment]
This more like a discussion about the usage, moving the issue to discussion to get more people to join the discussion.

@geantrindade - haven't seen your reply for few days, get any luck on trying the Random Strategy as @ultmaster mentioned lately?
